vmm/src/device_manager.rs

```diff
@@ -392,6 +410,50 @@ pub fn get_win_size() -> (u16, u16) {
     (ws.cols, ws.rows)
 }
 
+#[cfg(any(target_env = "musl"))]
+const TIOCSPTLCK: libc::c_int = 0x4004_5431;
+#[cfg(any(target_env = "musl"))]
+const TIOCGTPEER: libc::c_int = 0x5441;
+
+#[cfg(any(target_env = "gnu"))]
+const TIOCSPTLCK: libc::c_ulong = 0x4004_5431;
+#[cfg(any(target_env = "gnu"))]
+const TIOCGTPEER: libc::c_ulong = 0x5441;
+
+pub fn create_pty() -> io::Result<(File, File)> {
+    let main = match OpenOptions::new()
+        .read(true)
+        .write(true)
+        .custom_flags(libc::O_NOCTTY)
+        .open("/dev/pts/ptmx")
+    {
+        Ok(f) => f,
+        _ => OpenOptions::new()
+            .read(true)
+            .write(true)
+            .custom_flags(libc::O_NOCTTY)
+            .open("/dev/ptmx")?,
+    };
+    let mut unlock: libc::c_ulong = 0;
+    unsafe { libc::ioctl(main.as_raw_fd(), TIOCSPTLCK, &mut unlock) };
```

Comment:
The prototype difference between ioctl on musl vs glibc is a known issue. This difference is thus reflected in the Rust bindings. Is it possible to define the constants as i32 and then use .into() which would be a no op on musl and a safe extension from i32 to u64?

Reply:
`.into()` wasn't happy because of signedness but `try_into().unwrap()` seems happy enough and should never fail given the values (though no longer a no-op but this isn't a hot path). Thanks!